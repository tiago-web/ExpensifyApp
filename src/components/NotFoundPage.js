import React from "react";
import { useLocation, Link } from "react-router-dom";

export default function NotFoundPage() {
	let location = useLocation();

	return (
		<div>
			No match for <code>{location.pathname}</code> <Link to="/">Go home</Link>
		</div>
	);
}
