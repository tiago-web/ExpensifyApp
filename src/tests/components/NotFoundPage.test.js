import React from "react";
import { shallow } from "enzyme";
import toJSON from "enzyme-to-json";
import NotFoundPage from "../../components/NotFoundPage";

jest.mock("react-router-dom", () => ({
	...jest.requireActual("react-router-dom"),
	useLocation: () => ({
		pathname: "localhost:3000/example/path",
	}),
}));

test("should render NotFoundPage correctly", () => {
	const wrapper = shallow(<NotFoundPage />);

	expect(toJSON(wrapper)).toMatchSnapshot();
});
