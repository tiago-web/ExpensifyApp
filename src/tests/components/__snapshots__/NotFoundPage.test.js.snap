// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render NotFoundPage correctly 1`] = `
<div>
  No match for 
  <code>
    localhost:3000/example/path
  </code>
   
  <Link
    to="/"
  >
    Go home
  </Link>
</div>
`;
